### PR TITLE
ci: build samples/cpp/hello_world as part of the multiplatform test

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Build the C++ version of the Hello, World sample as part of the multiplatform (build) test in CI.

This was inspired by https://github.com/zephyrproject-rtos/zephyr/issues/78263

### CODE REVIEW

File: .github/workflows/hello_world_multiplatform.yaml [hunk 1]
```diff
@@ -29,6 +29,12 @@ jobs:
         os: [ubuntu-22.04, ubuntu-24.04, macos-13, macos-14, windows-2022]
     runs-on: ${{ matrix.os }}
     steps:
+      - name: Install packages
+        if: startsWith(runner.os, 'Linux')
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y g++-multilib
```
Comment (fabiobaltieri) [inline .github/workflows/hello_world_multiplatform.yaml:36]: should this go straight in `https://github.com/zephyrproject-rtos/action-zephyr-setup/blob/main/action.yml#L48`? May save some hassle for c++ user for not much inconvenience for everyone else

**Comment by henrikbrixandersen** [comment]
The failing tests on `windows-2022` and `macos-13` seem to point to actual issues with the Zephyr SDK, as described in https://github.com/zephyrproject-rtos/zephyr/issues/78263.

File: .github/workflows/hello_world_multiplatform.yaml (hunk 1 above)
Comment (henrikbrixandersen) [reply .github/workflows/hello_world_multiplatform.yaml:36]: Probably, yes. I only added it here to be able to verify the builds. We can add it to the setup action and then remove it from here when bumping the version of that action?

Comment (fabiobaltieri) [reply .github/workflows/hello_world_multiplatform.yaml:36]: If you open a PR for that I can merge it and point the v1 tag to it right now, no need to bump it.

Comment (henrikbrixandersen) [reply .github/workflows/hello_world_multiplatform.yaml:36]: I've opened https://github.com/zephyrproject-rtos/action-zephyr-setup/pull/19

**Review by fabiobaltieri — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Comment by stephanosio** [comment]
Regarding the non-Linux build failures, please see https://github.com/zephyrproject-rtos/zephyr/issues/78263#issuecomment-2359868714.

**Review by dkalowsk — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
This depends on https://github.com/zephyrproject-rtos/zephyr/pull/80192

**Comment by stephanosio** [comment]
@henrikbrixandersen Rebasing should fix the CI failure now.

**Comment by henrikbrixandersen** [comment]
> @henrikbrixandersen Rebasing should fix the CI failure now.

Thanks, rebased on `main`.

**Review by simhein — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**